index.c

```diff
@@ -2572,6 +2572,24 @@ int mutt_index_menu(void)
           menu->redraw = REDRAW_CURRENT;
         break;
 
+      case OP_ADD_ID:
+        if (!prereq(Context, menu, CHECK_IN_MAILBOX | CHECK_MSGCOUNT | CHECK_VISIBLE | CHECK_ATTACH))
+          break;
+        mutt_add_content_id(CUR_EMAIL, CUR_EMAIL->content, NULL);
```

Comment:
Ah!  Since you created your branch, `CUR_EMAIL` (an evil macro) has been eliminated.
You can replace it, with:

```c
struct Email *e_cur = get_cur_email(Context, menu);
```